gradoop-common/src/main/java/org/gradoop/common/model/impl/properties/strategies/PropertyValueStrategyFactory.java

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright © 2014 - 2019 Leipzig University (Database Research Group)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradoop.common.model.impl.properties.strategies;
+
+import org.gradoop.common.model.impl.id.GradoopId;
+import org.gradoop.common.model.api.strategies.PropertyValueStrategy;
+
+import java.math.BigDecimal;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Factory class responsible for instantiating strategy classes that manage every kind of access to
+ * the current {@code PropertyValue} value.
+ */
+public class PropertyValueStrategyFactory {
+
+  /**
+   * {@code PropertyValueStrategyFactory} instance
+   */
+  private static PropertyValueStrategyFactory INSTANCE = new PropertyValueStrategyFactory();
+  /**
+   * Map which links a data type to a strategy class
+   */
+  private final Map<Class, PropertyValueStrategy> classStrategyMap;
+  /**
+   * Map which link a type byte to a strategy class
+   */
+  private final Map<Byte, PropertyValueStrategy> byteStrategyMap;
```

Comment:
Ok yes replacing the map with an array would be possible, but im not sure i get what you are after here. Do you think using an array would result in a better performance?

Reply:
The type bytes are all small byte values, replacing the Map with just `PropertyValueStrategy[]` using the type bytes as indices will reduce the look-up time from logarithmic to constant. This won't have a small effect, but since those strategies are looked up every time a propery value is accessed (right?), it should have a positive effect.

Reply:
What is the opinion here? I think the approach of @p-f is worth a try.

Reply:
I implemented the byteStrategyMap as `PropertyValueStrategy[]`